Adds Syndicate Listening Post

DangerRevolution opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->
First-step of adding Syndicate Listening Post mid-round ghostrole
Left to do:

- [X] Add the shuttle itself
- [X] Syndicate Radio Operator - Ghost Role
- [X] Syndicate Guards - AI Antagonists (Thanks to @evilexecutive)
- [X] Syndicate AA Radio 
- [X] Repurpose code to make the radio station spawn mid-round
- [x] Localise ghost text for Listening Operator
- [X] Add Syndie Crew Monitor
- [x] Recreate the shuttle to fit new additions as mentioned above and Decorate!!

## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->
It was a fun ghostrole in Station 13, however being:
A) A ghostrole in Station 13
B) Positioned 4KM away, in a derelict outpost with nothing but cams and a comms server to look at; as well as your PDA was a bit mediocre.

This brings in a semi-mobile disguised operations base; included with a crew monitor and a fax machine, as well as two AI Guards.

This is; frankly, a RolePlay role. Especially with no guarantee there'll been be Syndies in the AO, it'll be up to the players onboard to guide what happens with the shuttle, and how much they interfere, and how much they monitor.

This also maps in the Syndie Fax as well as several stamps, plus some self-defence weapons to the shuttle.

It also adds something for Salvage / Security to seize.
## Technical details
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
Mostly a .yml addition with a rehash of LoneOP and Ninja code to let it spawn mid-round, as well as a new crew monitoring server to pick up suit sensors within 2KM (to be added).
## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
I have added the first draft of the shuttle.
![image](https://github.com/DeltaV-Station/Delta-v/assets/142105406/6ea4e808-65e4-4092-815c-9b6cd7cc1fc6)
<!--
List any breaking changes, including namespace, public class/method/field changes, prototype renames; and provide instructions for fixing them. This will be pasted in #codebase-changes.
-->

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->
:cl: DangerRevolution and VMSolidus
- add: Rumours say that Syndicate Agents have cracked encoded communications! Added the Syndicate Listening Post to the Syndicate's arsenal of espionage!

DangerRevolution committed:
added listening post

DangerRevolution committed:
adds code for radio station to spawn mid-round

DangerRevolution committed:
updated events.yml

DangerRevolution committed:
tweaked pirate radiostation

DangerRevolution committed:
Added Syndicate Listener

I can't get it to spawn when I "request" it via ghost role, nor the entity spawner but it's a start...

DangerRevolution committed:
Finished Listener Ghost Role and Radio

DangerRevolution committed:
Changes Listening Agent's Items

DangerRevolution committed:
Merge branch 'master' into master

DangerRevolution committed:
misc gear fixes + shuttle fixes

DangerRevolution committed:
Merge branch 'master' of https://github.com/DangerRevolution/SyndicateListeningPost

DangerRevolution committed:
fixed mistake in nukie starting gear

DangerRevolution committed:
Allows RadioOp to recieve codewords but not objectives (Thanks to @Ps3Moira)

DangerRevolution committed:
Localised ListeningOp ghost role

VMSolidus commented:
Oh god you put the entities in wizden's namespace. Uhh I'll go ahead and move them to DeltaV namespace and make appropriate files.

VMSolidus commented:
![image](https://github.com/DeltaV-Station/Delta-v/assets/16548818/73f16467-3fbb-4c25-b190-9a63fe9b2f36)
I'm going to fix a bunch of this too. We don't need to comment or modify wizden code.

ps3moira commented:
> Oh god you put the entities in wizden's namespace. Uhh I'll go ahead and move them to DeltaV namespace and make appropriate files.

:trollface:

DangerRevolution commented:
> ![image](https://private-user-images.githubusercontent.com/16548818/288962390-73f16467-3fbb-4c25-b190-9a63fe9b2f36.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIwMTA2ODQsIm5iZiI6MTcwMjAxMDM4NCwicGF0aCI6Ii8xNjU0ODgxOC8yODg5NjIzOTAtNzNmMTY0NjctM2ZiYi00YzI1LWIxOTAtOWE2M2ZlOWIyZjM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA4VDA0Mzk0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3NmRiZmZmYzU0M2FjOTYwM2U1M2Y1YWNhMjllNDczZjkyMDg4MjA1NDBhNjJmMjc4ODU5ZjQ1ODljMzNlZWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8QSkmx7FIQ5KGpbk6z1PfvAH_QJi20i98mYuddVrsko) I'm going to fix a bunch of this too. We don't need to comment or modify wizden code.

I knew that I'd have to sort it out into DeltaV folders but it started off as trial and error and after the hours it took me to get it sorted I was too lazy to go back over it and move it into DeltaV folders. I appreciate it though :)
the comments are also leftover parts of experimenting with the existing code; that again, hours of tedium made me too lazy to take out. I can passover and move things tomorrow (afternoon GMT time).

VMSolidus commented:
https://github.com/DeltaV-Station/Delta-v/assets/16548818/bcc39d67-6c45-4d74-a091-8910aa6c34a9

But that's okay because here's the radio guards, I'll make their spawner next and then commit! They have the same ammo capacity as a standard Viper, and infact they shoot the same bullets as a Viper. Sadly it's not possible to have them once every 5 seconds reload the entire magazine. But there's also a couple different EntityAmmoProviders, so I'm going to experiment with a few and see what options I like. 

As you can see, these assholes are insanely aggressive. And yes, he did infact pick up and fire an RPG at the end of that video. If they see a gun they like that is better than their stock gun, they WILL pick it up and use it.

ps3moira commented:
> Recording.2023-12-07.234322.mp4 
> But that's okay because here's the radio guards, I'll make their spawner next and then commit! They have the same ammo capacity as a standard Viper, and infact they shoot the same bullets as a Viper. Sadly it's not possible to have them once every 5 seconds reload the entire magazine. But there's also a couple different EntityAmmoProviders, so I'm going to experiment with a few and see what options I like.

Can you make fire the gun at a slower rate? Also for custom Sprites for headset and PDA's there's already syndicate headsets and newly added Syndicate Agent PDA's.

VMSolidus commented:
Yes, I can arbitrarily control their fire rate, and decrease it(I probably will, since maxing out the fire rate of a Viper is INSANE).

DangerRevolution commented:
@evilexecutive 
When you commit I'd appreciate if you could take a quick walk around it? Just take the ghost role and look around. Syndie Cams and Crew Monitor are a *soon*, I think I've found how to make them read across grids.

VMSolidus commented:
![image](https://github.com/DeltaV-Station/Delta-v/assets/16548818/086f7bac-4ff2-4d73-9352-d71a7e52e175)
Not too bad, although I think we should definitely replace the standard reinforced walls with the same walls that the Nukie shuttle is made with. That'll visually read to salvagers who encounter it that "This is extremely fucking dangerous", especially when they hear the boots walking around inside it from the NPC guards that randomly wander its halls

VMSolidus commented:
![image](https://github.com/DeltaV-Station/Delta-v/assets/16548818/c4eb96b1-7eb4-4d3d-958c-ca7891499ff9)
Alright while I was at it, I also took a crack at "Prettying up" the outpost. Adding its asteroid camouflage, replacing the NT standard reinforced walls with Syndicate standard Plastitanium. I replaced the window with syndicate standard plasma glass with blast doors(that can be shut by the radio operator on his bridge. I also added sleeping quarters for the outpost guards. My version of the map is saved as pirateradiosolidus.yml, I didn't overwrite yours because I figured you would probably want to take a look at my version and make your own changes before making a "Main" version of the map.

VMSolidus committed:
I LOST THE MAP CHANGES AAAAAAAAAAAAAAAA

VMSolidus committed:
Alright I found it and remade it

VMSolidus commented:
![image](https://github.com/DeltaV-Station/Delta-v/assets/16548818/194af9f8-cc79-46eb-b95a-eb61e832caae)
Awesome, the map loads in correctly now, and the guard spawners WORK

VMSolidus commented:
There's still more work to do on the map, but I'll go do that tomorrow after I've had some sleep. I need to make the map more practical for "Long Term Observation", so it needs an extremely minimalist Atmos, a Rec Room, Tool Room, and a Minimalist Botany. I already know where I'm going to put all 4 of those things.

VMSolidus committed:
I may have ended up finishing it anyway

VMSolidus commented:
![image](https://github.com/DeltaV-Station/Delta-v/assets/16548818/75485f5d-842f-4f46-8607-aee375255d4e)
I may have accidentally finished the outpost (except for decoration stuff) instead of sleeping.

VMSolidus committed:
Nerfing their fire rate down to something actually reasonable

DangerRevolution commented:
> ![image](https://private-user-images.githubusercontent.com/16548818/289016740-75485f5d-842f-4f46-8607-aee375255d4e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIwMjg5MzUsIm5iZiI6MTcwMjAyODYzNSwicGF0aCI6Ii8xNjU0ODgxOC8yODkwMTY3NDAtNzU0ODVmNWQtODQyZi00ZjQ2LTg2MDctYWVlMzc1MjU1ZDRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA4VDA5NDM1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwYjVhYWQzM2M1MjExZjU5YzI0MWMzMzE0Njc4YzE1MzIwNzJlOTYwZTBlMWZmYTk3ODJmNDg2ZjYxY2VjNDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Zg7uJc_dnf8S7YlVm7E69aM5A-AQo-d_L3pufE2dPLE) I may have accidentally finished the outpost (except for decoration stuff) instead of sleeping.

Drippyyy I can't wait to see this. Just woke up but I'll get back to you.

DangerRevolution committed:
Removed outdated code comments. Fixed localisation.  Transferred things to be marked as Delta or in the Delta folder.

DangerRevolution committed:
Localised Localisation

DangerRevolution committed:
Create syndicate_monitor_server.yml

DangerRevolution commented:
[8162819](https://github.com/DeltaV-Station/Delta-v/pull/526/commits/8162819e67cb1d0d5479633e6ba8506d474edab1)
I've actually managed to make the Syndie Crew Monitoring Server; it picks up Suit Sensors past the range of the *Crew* Monitoring Server. Cameras themselves are hardcoded to *not* broadcast past 200. Might just up the broadcast range on the Cameras themselves as this shouldn't impact... anything, the monitor and the server won't pick it up anyways as it's not in range.

BUT if it goes badly this could break cameras on every map...

DangerRevolution committed:
changes standard shuttle to be loaded to Solidus' version

DangerRevolution commented:
Adding a camera monitor to the shuttle would require some creative thinking. normal cameras can only be read by the router if they're on the same grid, the monitor can only read the router on the same grid.
wireless camera routers don't have this issue; but they're also not very useful.

I've changed locales so far, added the crew monitor under "suspicious crew monitoring server", I'll map-in the changes and some extra additions + thrusters + I'll get an Atmos main to look at the map in about 6 hours from now.

DangerRevolution committed:
Got the Syndie Names to Work!!

DangerRevolution commented:
Changed inheritance so now the Listener Ops inherit names.

All that's left really is to drip the shuttle out more and add thrusters; which I'll do within the 6 hours and have a look at letting any race spawn there. Only concern is how many minutes of functional Oxygen the shuttle has..

Down the line; Syndie Cam Monitor would probably require upstream changes to wired/wireless; more listening-op specific drip would be nice.

VMSolidus commented:
According to my math, the listening outpost has enough air to last its two radio operators 39.3 hours.

DangerRevolution committed:
updated localisation

VMSolidus committed:
The map is now finished!

VMSolidus commented:
![syndicatelisteningpost](https://github.com/DeltaV-Station/Delta-v/assets/16548818/6742203a-244a-4cc1-a7e5-10d81c6cec1f)
The actual map for the Syndicate Listening Post is now done!

DangerRevolution committed:
Delete pirateradio.yml

VMSolidus committed:
Go!

VMSolidus committed:
Update pirateradio.yml

DangerRevolution committed:
Updated the Radio Operator .yml

DangerRevolution committed:
updated the listening post!

VMSolidus committed:
Fixing those errors

VMSolidus committed:
Update headsets.yml

DangerRevolution committed:
Fixing errors.

DangerRevolution committed:
Localised PirateSpawnRules

DangerRevolution committed:
Delta-ised human.yml Syndicate Listener

DangerRevolution committed:
Delta-ised humanoid.yml

DangerRevolution committed:
Delta-ised events.yml

DangerRevolution committed:
Fixed a weird space.

DangerRevolution committed:
Fixed spelling mistake >:(

DangerRevolution committed:
Fixed an error.

DangerRevolution committed:
Update humanoid.yml

DangerRevolution committed:
Another error!!

DangerRevolution committed:
Merge branch 'master' into master

VMSolidus committed:
Update human.yml

VMSolidus committed:
Update pirateradio.yml

DangerRevolution committed:
final fixes on pirateradio.yml

VMSolidus committed:
keep guards from picking up guns

DangerRevolution committed:
Fixed crash to desktop upon shooting

VMSolidus committed:
Apparently I cant do that

DangerRevolution committed:
fixed bathroom light

DangerRevolution committed:
Merge branch 'master' of https://github.com/DangerRevolution/SyndicateListeningPost

DangerRevolution commented:
This PR is like; done done. We're happy with it's state and happy for it to be reviewed.

DangerRevolution committed:
Fixed stray space.

DangerRevolution commented on Resources/Locale/en-US/deltav/ghost/roles/ghost-role-component.ftl:
```diff
@@ -2,4 +2,7 @@ ghost-role-information-nukie-mouse-name = Nuclear Operative Mouse
 ghost-role-information-nukie-mouse-description = A Nuclear Operative reinforcement for the Syndicate.
 ghost-role-information-nukie-mouse-rules = Normal syndicate antagonist rules apply. Work with whoever called you in, and don't harm them.
                                     The crew is allowed to kill you without warning.
-                                    You are allowed to attack the crew and destroy the station without provocation. 
```
Can't figure out how to fix this >:(
the text is still there!!

VMSolidus replied:
I gotcha, the secret is you have to use a program other than notepad.

VMSolidus committed:
Update ghost-role-component.ftl

VMSolidus committed:
Update ghost-role-component.ftl

VMSolidus committed:
Update pirateradio.yml

IamVelcroboy commented:
Map checks out. I just have 1 request: Map some pajamas in. (Alternatively, we could add them to the SyndieVend but either way, they should have them here 🙂)

Other thing of note is that some of the internal airlocks are not "locked" variants. That's ok if intentional. Just wanted to point it out if you did not intend for that.

DangerRevolution commented:
> Other thing of note is that some of the internal airlocks are not "locked" variants. That's ok if intentional. Just wanted to point it out if you did not intend for that. 

A) some of the non-essential locks are glass and unlocked due to give salvage a little leeway; the airlocks are ROBUST and it'll be tedious to break every single door down. The entrance locks are locked to give Radio Ops time to spring their trap, I'm open to @evilexecutive 's opinion though!

> Map checks out. I just have 1 request: Map some pajamas in. (Alternatively, we could add them to the SyndieVend but either way, they should have them here 🙂

B) The radio operators are dripped in the Pyjama's, so there's at least one pair guaranteed. I can add a spare in the bedrooms if you meant a *mapped* version of it?

VMSolidus committed:
finishing touches

IamVelcroboy commented:
> A) some of the non-essential locks are glass and unlocked due to give salvage a little leeway; the airlocks are ROBUST and it'll be tedious to break every single door down. The entrance locks are locked to give Radio Ops time to spring their trap, I'm open to @evilexecutive 's opinion though!

Works for me! 

> B) The radio operators are dripped in the Pyjama's, so there's at least one pair guaranteed. I can add a spare in the bedrooms if you meant a _mapped_ version of it?

Radio operators are the ghost role? Not one of the AIs? 
If so, then all good. Don't really need extra then.

VMSolidus commented:
> Map checks out. I just have 1 request: Map some pajamas in. (Alternatively, we could add them to the SyndieVend but either way, they should have them here 🙂)

![image](https://github.com/DeltaV-Station/Delta-v/assets/16548818/0fe34930-b445-4343-bcd9-262dd891e459)
So the radio operators spawn wearing comfortable Pyjamas, but it was an oversight that the guards have nothing comfortable to sleep in. I've given the guard sleeping quarters a bag for their PJs. 

![image](https://github.com/DeltaV-Station/Delta-v/assets/16548818/99a6c7ca-c482-4cc1-8d25-9c2c2d3b19fc)
I also reorganized the door locks and moved the door to the botany to instead come from the Rec Room. The only locked doors in the facility are now the "Secure" doors. The Entrance, Bridge, Armory, "Atmos", and power room are all locked doors now, but all other doors are no longer locked.

Colin-Tel commented on Resources/Prototypes/DeltaV/GameRules/events.yml:
```diff
@@ -45,3 +45,16 @@
+    earliestStart: 20
+    weight: 7.5 # Worth some thought as to how frequent this should be...
+    minimumPlayers: 25
+    reoccurrenceDelay: 180
```
```suggestion
    maxOccurrences: 1
```
This should perform the function of only having it happen once per round.

DangerRevolution replied:
thanks!
i've kept reocurrence delay in-case MaxOccurences fails

DangerRevolution committed:
final changes + playtime setting for final review!

DangerRevolution committed:
fix spelling error

DangerRevolution committed:
Merge branch 'master' into master

Colin-Tel commented on Resources/Prototypes/DeltaV/Entities/Clothing/Ears/headset_alt.yml:
Why was this deleted?

DangerRevolution replied:
I don't know but the HoP headset within headset_alt.yml is contained within headset.yml; same folder just in a different file name

Colin-Tel replied:
Oh I didn't even see you moved it. Why didn't you just say you moved it lol

DangerRevolution replied:
i didn't know :shrug: :smile:

Colin-Tel commented on Resources/Prototypes/DeltaV/GameRules/events.yml:
```diff
@@ -45,3 +45,17 @@
+    weight: 7.5 # Worth some thought as to how frequent this should be...
+    minimumPlayers: 25
+    maxOccurences: 1
+    reoccurrenceDelay: 180 #fallback for maxoccurences failing
```
Why would maxOccurrences fail?

DangerRevolution replied:
I don't know but I like to be safe, I can remove it if necessary!

Colin-Tel commented on Resources/Locale/en-US/deltav/ghost/roles/ghost-role-component.ftl:
```diff
@@ -1,5 +1,8 @@
+                                           You are allowed to attack the crew and destroy the station without provocation. 					
+ghost-role-information-listeningop-name = Listening Post Operative
+ghost-role-information-listeningop-description = You are a Listening Post operative. Get into range, observe the station, intercept communications and assist any operatives in the area!
+ghost-role-information-listeningop-rules =	You are a Syndicate Operative tasked with the continous reporting and monitoring of the station, it's activities and of assisting any fellow operatives who may be aboard the station. As an antagonist, do whatever is required to complete this task.  Do NOT abandon your station, and make sure it does not fall into enemy hands! Hide your existence at any co ...
```
```suggestion
ghost-role-information-listeningop-rules =	You are a Syndicate Operative tasked with the continuous reporting and monitoring of the station and its activities, as well as assisting any fellow operatives who may be aboard the station. As an antagonist, do whatever is required for you to complete this task. Make sure your station doesn't fall into enemy hands and DO NOT abandon your station! Hide your existence at any cost!
```

VMSolidus committed:
Update Resources/Locale/en-US/deltav/ghost/roles/ghost-role-component.ftl

Co-authored-by: Colin-Tel <113523727+Colin-Tel@users.noreply.github.com>
Signed-off-by: VMSolidus <evilexecutive@gmail.com>

VMSolidus committed:
Update events.yml

Signed-off-by: VMSolidus <evilexecutive@gmail.com>

DangerRevolution committed:
Merge branch 'master' into master

DangerRevolution committed:
Merge branch 'master' into master

DangerRevolution commented:
Closed to make use of uh, branches; reopened as #578
